[18.0][MIG] excel_import_export: Migration to 18.0

xaviedoanhduy opened the pull request:
- remove useless `view_init` in commit: https://github.com/odoo/odoo/pull/79563/commits/fbfbf1756b
- fix domain ([issue](https://github.com/OCA/server-tools/pull/2991#discussion_r1791811592)) on `template_id` field of import/export wizard. for some reason `context` is no longer present in python dynamic domains, see example here: https://github.com/odoo/odoo/pull/135145/commits/6828dd3489
- user service is no longer a service, see: https://github.com/odoo/odoo/pull/145362/commits/ae660431

kittiu committed:
Excel Import/Export/Report  (#1522)

* [ADD] v12 excel_import_export

* Change from eval() to safe_evel()

* Change variable to format to style, as fomat is a common python function

:100644 100644 00ee3d9f... e9e48d87... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 01e5b9f5... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 17d3964d... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 933ce0dc... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 3b1217e8 M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 a7d6adc5 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 e3826e08 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 c8481487 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 ed8c9fc7 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 69aa6ea0 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 c9733b95 A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 bb3ea32e A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

kittiu committed:
[12.0][FIX] excel_import_export

wrong use of dict.iteritems() for python3

OCA-git-bot committed:
[ADD] icon.png

kittiu committed:
[12.0][FIX] excel_import_export

Fix Import Excel wizard, cannot download template as in v12
core Odoo prevent download default binary.

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][FIX] excel_import_export, use sudo() to get param

kittiu committed:
[12.0][IMP] excel_import_export, excel_import_export_demo

Add report action feature and new examples

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][FIX] excel_import_export, fix wrong file name on report_action

When mark to convert xlsx to csv, csv file name is not used

Saran440 committed:
[12.0][FIX] excel_import_export - Style sum footer to number

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 2.9% (5 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 24.1% (41 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (170 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (170 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][ENH] excel_import_export, faster export

Previously export is slow on large number of rows.
This fix ensure that sheet.insert_rows is called only once for all rows,
instead of insert 1 row for every rows

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

kittiu committed:
[12.0][FIX] excel_import_export, problem on import with datetime field

Can't import if the import file has date time field

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

kittiu committed:
[IMP] : black, isort, prettier

kittiu committed:
[13.0][MIG] excel_import_export, excel_import_export_demo

kittiu committed:
[IMP] excel_import_export: black, isort, prettier

kittiu committed:
[14.0][MIG] excel_import_export, excel_import_export_demo

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/

alessandro-fiorino committed:
Added translation using Weblate (Italian)

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 5.5% (11 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

OCA-git-bot committed:
[IMP] update dotfiles

Commit:
[15,0][MIG] excel_import_export, excel_import_export_demo:

migration to 15.0. Changes by pre-commit errors and warnings,
changes by original Odoo code changes, JS files suggestions.

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

luisg123v committed:
[REF] excel_import_export: remove xlrd & xlwt from external dependencies

Because those dependencies are already part of Odoo's base dependencies.

OCA-git-bot committed:
excel_import_export 14.0.1.1.0

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (200 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/es_AR/

Commit:
[15.0][FIX] excel_import_export, excel_import_export_demo:

fix from PR #2283 together with @luc-demeyer suggestion https://github.com/OCA/server-tools/pull/2283#pullrequestreview-881026019

AungKoKoLin1997 committed:
[MIG] excel_import_export, excel_import_export_demo: Migration to 16.0

AungKoKoLin1997 committed:
[FIX] Not overriding the create method during installation

Commit:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 81.1% (155 of 191 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.1% (155 of 191 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.1% (155 of 191 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Commit:
[UPD] Update excel_import_export.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (192 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/es/

Commit:
[UPD] Update excel_import_export.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

RodrigoBM committed:
[REF] excel_import_export: hooks are added so that they can be inherited

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

OCA-git-bot committed:
[BOT] post-merge updates

PauBForgeFlow committed:
[IMP] excel_import_export: Set import wizard filename

Commit:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (193 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 80.8% (156 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 86.5% (167 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (193 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (193 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Commit:
[UPD] Update excel_import_export.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (194 of 194 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (194 of 194 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (194 of 194 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Commit:
[UPD] Update excel_import_export.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (203 of 203 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (203 of 203 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (203 of 203 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

AungKoKoLin1997 committed:
[FIX] excel_import_export: remove sudo() when importing record

This commit fixes issue of finding multiple records across all companies during import when they have same record name.
Previously, importing with sudo could bypass multi-company record rules.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (203 of 203 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[IMP] excel_import_export: improve error showing

Commit:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (204 of 204 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

AungKoKoLin1997 committed:
[IMP] excel_import_export: excel_import_export: allow selection of all related templates for export action

Before this commit, when two templates exist for the same model, clicking "ADD EXPORT ACTION" for each template
creates separate server actions for each. As a result, if multiple templates are created for the same model,
numerous server actions with the label "Export Excel" will accumulate for that model.
With this commit, if two templates exist for the same model and "ADD EXPORT ACTION" is clicked for each,
they will now share a single server action. Users can then select templates from within the "Export Excel" wizard.
Clicking "REMOVE EXPORT ACTION" on a template will remove only that template from the "Export Excel" selection.
The server action itself will be deleted only when no templates use it for export.

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[IMP] excel_import_export: update readme

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (204 of 204 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

OCA-git-bot committed:
[BOT] post-merge updates

xaviedoanhduy committed:
[IMP] excel_import_export: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] excel_import_export: Migration to 18.0

wlin-kencove commented:
Testing locally and here are some things I have noticed:

It looks like the export function works, though the files are created without the option to set a file name. Instead the files are named based on their file size. The more problematic issue is that the files are named without their extensions:

<img width="751" alt="image" src="https://github.com/user-attachments/assets/817fb69f-1a61-47e0-b058-d9b095503452" />

When opened the files seem to be correct but they cannot be used for imports:
<img width="892" alt="image" src="https://github.com/user-attachments/assets/e96d36aa-f1ca-4793-b997-2b27c838b4af" />

Error on trying to import file:
<img width="1205" alt="image" src="https://github.com/user-attachments/assets/28f4b71d-01b4-4a20-b152-a038c76bb043" />

wlin-kencove commented:
Manually adding .xlsx does not fix the import error 

<img width="1061" alt="image" src="https://github.com/user-attachments/assets/b74db606-e366-4af8-bc9a-e3b7e6cf15d8" />


<img width="1323" alt="image" src="https://github.com/user-attachments/assets/907700ce-32d8-458a-929c-79e82956ad84" />

xaviedoanhduy commented:
hi @wlin-kencove, fixed the above errors.

wlin-kencove commented:
The export looks good now, though the import is still displaying errors for me:
<img width="1225" alt="image" src="https://github.com/user-attachments/assets/272cc8bd-1ba6-45f1-8d93-a6bbd4291485" />


I am using the file exported without editing it.  I have it attached if it helps:
[S00001.xlsx](https://github.com/user-attachments/files/18967889/S00001.xlsx)

wlin-kencove commented:
<img width="1002" alt="image" src="https://github.com/user-attachments/assets/6c21c309-6efd-4d11-b3df-a99525f61eca" />

This is what I am passing.


This is the error
<img width="1353" alt="image" src="https://github.com/user-attachments/assets/d1531d75-d347-42d9-a140-0b7e79a1b6bb" />

This is the file. 
[S00001.xlsx](https://github.com/user-attachments/files/18968698/S00001.xlsx)

xaviedoanhduy commented:
strange, i tried importing your file by renaming the product and removing the tax field. but thanks for raising the issue. there might be a module that is affecting this feature. if possible please upgrade the module again and in the meantime i will try to investigate.

[Screencast from 25-02-2025 23:44:23.webm](https://github.com/user-attachments/assets/ff476d74-8aba-423d-9be2-480da3e95fab)

xaviedoanhduy commented:
hi @wlin-kencove, i found out where the problem is. the error is in the `xlrd` library in the new version (which is no longer available with .xlsx file type), see: https://github.com/python-excel/xlrd/blob/0c4e80b3d48dfe2250ac4e514c8231a742fee221/xlrd/__init__.py#L138

and i will try to make this module compatible with new version of `xlrd`, please be patient

xaviedoanhduy committed:
[IMP] excel_import_export: replace reading excel file from xlrd with openpyxl

[REF] excel_import_export: refactor _import_record_data for better readability and maintainability

wlin-kencove commented:
> hi @wlin-kencove, i found out where the problem is. the error is in the `xlrd` library in the new version (which is no longer available with .xlsx file type), see: https://github.com/python-excel/xlrd/blob/0c4e80b3d48dfe2250ac4e514c8231a742fee221/xlrd/__init__.py#L138
> 
> and i will try to make this module compatible with new version of `xlrd`, please be patient

Ah I see, versions before  2.00 could read xlsx but afterwards it only reads xls. 
Our difference experiences must be due to the difference in versions. 
Maybe something like openpyxl could be a replacement?

xaviedoanhduy commented:
hi @wlin-kencove, i added some updates to fix import error with `xlrd`. can you check again?

wlin-kencove commented:
<img width="1398" alt="image" src="https://github.com/user-attachments/assets/67931b35-88ec-4ca4-8aae-67b9b8ce38f3" />

@xaviedoanhduy 

Looks like that did it works for me now. Thanks for taking a look!

BhaveshHeliconia commented:
@xaviedoanhduy :
Test Case 1:
- Try to update "account.move" record using following configuration("Settings > Technical > Excel Import/Export > XLSX Templates")  using attached excel template file. 
![1](https://github.com/user-attachments/assets/5a195047-6f93-491f-a407-b6b7181c5582)

![2](https://github.com/user-attachments/assets/90f1c665-d22c-4891-91ee-86b65273d852)

[test_import_template.xlsx](https://github.com/user-attachments/files/20757445/test_import_template.xlsx)

- Following error occurs when try to update "account.move" record("Accounting > Accounting > Journal Entries"): 
![3](https://github.com/user-attachments/assets/4f63676b-0d93-4efb-b637-83a06b3ac53f)

![4](https://github.com/user-attachments/assets/e555c72a-d5e6-4ef6-8ef8-c9259cc91727)

[test_import.xlsx](https://github.com/user-attachments/files/20757497/test_import.xlsx)

Test Case 2:
- Try to update "account.move" records using following configuration("Settings > Technical > Excel Import/Export > XLSX Templates")  using attached excel template file:
![5](https://github.com/user-attachments/assets/f6c47afe-2f94-4f68-a8c1-f207a5aa6289)

![6](https://github.com/user-attachments/assets/5b56b058-5421-494b-ad96-274eba551594)

[test_import_template_2.xlsx](https://github.com/user-attachments/files/20757510/test_import_template_2.xlsx)

- Field not set properly when try to update "account.move" record("Accounting > Accounting > Journal Entries"):
![7](https://github.com/user-attachments/assets/b45a2e55-dc61-41bc-88d7-f9d6bb48d41b)

![8](https://github.com/user-attachments/assets/82753c74-3ece-465c-bdad-52c16d562596)

![9](https://github.com/user-attachments/assets/44408d43-3f2b-44c3-b2bc-f72653de4225)

[test_import_2.xlsx](https://github.com/user-attachments/files/20757555/test_import_2.xlsx)

BhaveshHeliconia reviewed (changes requested):
Kindly check as mentioned in previous messages, can you please confirm where that issue is there or I am performing something wrong.

BhaveshHeliconia commented:
ping @xaviedoanhduy

xaviedoanhduy commented:
hello @BhaveshHeliconia, I see your format and configuration looks quite strange, you can refer to my following sample based on the information you want to import

In `Input Instruction (Dict.)` page
```
  {
                '__EXPORT__': {
                    'Sheet1': {
                        '_HEAD_': {
                            'B1': 'ref',
                            'B2': 'date',
                        },
                        'line_ids': {
                            'A5': 'account_id.display_name',
                            'B5': 'partner_id.display_name',
                            'C5': 'name',
                            'D5': 'credit',
                            'E5': 'debit',
                        }
                    }
                },
                '__IMPORT__': {
                    'Sheet1': {
                        '_HEAD_': {
                            'B1': 'ref',
                            'B2': 'date',
                        },
                        '_NODEL_line_ids': {
                           'A5': 'account_id',
                            'B5': 'partner_id',
                            'C5': 'name',
                            'D5': 'credit',
                            'E5': 'debit',
                        }
                    }
                },
                # '__POST_IMPORT__': '${object.post_import_do_something()}',
            }
```

the file content should be: 
![image](https://github.com/user-attachments/assets/86be2c5e-2f1d-4c93-bcb1-c3850d2577b4)

xaviedoanhduy commented:
input file should like this 
![image](https://github.com/user-attachments/assets/b90e0b41-8f10-415c-a87b-04bba52d37c0)

BhaveshHeliconia commented:
@xaviedoanhduy : I've tried to import with the sample described by you, but still it's not working. I've attached video showing steps I've performed and error and also attached excel files I've used in this video.

https://github.com/user-attachments/assets/c816f7fe-8737-4e49-ad70-6160b3254066

- [import_template.xlsx](https://github.com/user-attachments/files/21097797/import_template.xlsx)

- [import.xlsx](https://github.com/user-attachments/files/21097802/import.xlsx)

xaviedoanhduy commented:
@BhaveshHeliconia, do you have the same problem locally but on a different version? 
FYI, i checked on runboat and everything is working

![image](https://github.com/user-attachments/assets/ae055514-f158-405d-b2d7-e873b97aeb67)

BhaveshHeliconia commented:
@xaviedoanhduy : It's working fine now. I think it was missing this commit: https://github.com/OCA/server-tools/pull/3207/commits/fbbf238e70edb996901f60254ae8c345b54bc8d2

BhaveshHeliconia commented:
Functional test LGTM!!

Reyes4711-S73 reviewed (approved):
LGTM

Reyes4711-S73 commented:
@OCA/tools-maintainers Please, can you merge this PR?

thomaspaulb commented:
/ocabot merge nobump
